### PR TITLE
102-RefactorPaymentTypeView :: Created PaymentType view model

### PR DESCRIPTION
**Pull request by daynewright**
## Status

**READY**
## Documentation

[ ] I have fully documented features in this PR's code. Please put an `X` in the box to confirm.

[X ] This PR does **NOT** require documentation.
## Description

Adds view model for PaymentType view and updates the payment type view to use that model.
## Resolves Issue Number
#102
## Deploy Notes

Notes regarding deployment the contained body of work.  These should note any
db migrations, etc.

``` sh
git pull --prune
git checkout 102-RefactorPaymenTypeView
dotnet run
```
## Steps to Test or Reproduce

Outline the steps to test or reproduce the PR here.
1.  Run.
2. Select any user.
3. Navigate to the cart.
4. Click 'Select Payment Method'
5. Create payment method and submit.
6. Check database to see that payment method is created for the user in the PaymentType table.
## Impacted Files in Application

List general components of the application that this PR will affect:
- Controllers/PaymentTypesController.cs
- ViewModels/PaymentTypeCreate.cs
- Views/PaymentTypes/Create.cshtml

### CODE REVIEW

**Comment by daynewright** [comment]
@TeamCharles/team  Ready for review.

**Comment by garrett-vangilder** [comment]
LGTM Did throw an error if field in form is left blank.

File: Controllers/PaymentTypesController.cs [hunk 1]
```diff
@@ -27,17 +28,16 @@ public PaymentTypesController(BangazonContext ctx)
         public IActionResult Create()
         {
 
-        ViewBag.Users = Users.GetAllUsers(context);
+        var model = new PaymentTypeView(context);
```
Comment (MattKraatz) [inline Controllers/PaymentTypesController.cs:31]: Can you add this to the error validation on line 50 as well?

**Comment by Anulfo** [comment]
LGTM. The error is thrown with empty form or partially filled.

**Review by MattKraatz — CHANGES_REQUESTED** [review]

File: Controllers/PaymentTypesController.cs (hunk 1 above)
Comment (daynewright) [reply Controllers/PaymentTypesController.cs:31]: Should be fixed.

File: Controllers/PaymentTypesController.cs [hunk 2]
```diff
@@ -45,14 +45,9 @@ public IActionResult Create()
                 return RedirectToAction("Index", "Cart");
             }
             
-            ViewBag.Users = Users.GetAllUsers(context);
+            var model = new PaymentTypeView(context);
```
Comment (MattKraatz) [inline Controllers/PaymentTypesController.cs:48]: Add the existing PaymentType to this model before returning the view

**Review by MattKraatz — APPROVED** [review]

**Comment by MattKraatz** [comment]
LGTM once you commit the stuff we were looking at together

**Pull request merged**